### PR TITLE
GETTING_STARTED: misc usability improvements

### PR DESCRIPTION
**Pull request by feuGeneA**
I think it will be easier to review if you go commit by commit. Each commit is self-contained and quite simple.

### CODE REVIEW

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 1]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
-    bytes calldata message
-) external {
-    // Only the Teleporter receiver can deliver a message.
-    require(msg.sender == address(teleporterMessenger), "Unauthorized.");
-
-    // do something with message.
-}
+    /**
+     * @dev Receive a new message from another chain.
+     */
```
Comment (geoff-vball) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:229]: I don't think we need the dev tag on the example code

**Review by geoff-vball — COMMENTED** [review]
There's a lot here but generally LGTM. just left one nit

File: contracts/src/CrossChainApplications/GETTING_STARTED.md (hunk 1 above)
Comment (feuGeneA) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md:229]: It was already on some things, but not others, so I added it on everything else to be consistent. My only preference is that we be consistent. Would you prefer we skip the natspec entirely in this code?

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 2]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
-contract ExampleCrossChainMessenger is
+contract MyExampleCrossChainMessenger is
     ReentrancyGuard,
     ITeleporterReceiver
 {
+```
+```solidity
     using SafeERC20 for IERC20;
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:153]: The code blocks appear to be off here.

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 3]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
+                IERC20(feeTokenAddress),
+                feeAmount
+            );
+            IERC20(feeTokenAddress).safeIncreaseAllowance(
+                address(teleporterMessenger),
+                adjustedFeeAmount
+            );
+        }
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:180]: The code block indicators appear to be off here.

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 4]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
-});
-return
-    teleporterMessenger.sendCrossChainMessage(
-        TeleporterMessageInput({
+    function sendMessage(
+        ...
+    ) external returns (bytes32 messageID) {
+        ...
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:196]: Code blocks appear off

File: contracts/src/CrossChainApplications/GETTING_STARTED.md (hunk 1 above)
Comment (cam-schultz) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md:229]: If the goal is to replicate the ExampleCrossChainMessenger, I think it's alright to leave the natspec comments in.

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 5]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
+    /**
+     * @dev Receive a new message from another chain.
+     */
+    function receiveTeleporterMessage(
+        bytes32 sourceBlockchainID,
+        address originSenderAddress,
+        bytes calldata message
+    ) external {
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:236]: Code blocks appear off

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 6]
```diff
@@ -20,245 +20,277 @@ pragma solidity 0.8.18;
...
-        messageString
-    );
-}
+    function receiveTeleporterMessage(
+        ...
+    ) external {
+        ...
+        // do something with message.
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:265]: Code blocks appear off

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 7]
```diff
@@ -277,44 +309,59 @@ To start, replace the import for `ITeleporterReceiver` with `TeleporterOwnerUpgr
...
+```
+
+And at the beginning of `sendMessage()` add a call to get the latest `ITeleporterMessenger` implementation from `TeleporterRegistry`:
+
+```solidity
+    function sendMessage(
+        ...
+    ) external returns (bytes32 messageID) {
+```
+```solidity
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:344]: Code blocks appear off

File: contracts/src/CrossChainApplications/GETTING_STARTED.md (hunk 1 above)
Comment (geoff-vball) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md:229]: I don't really care that much, but does seem like we don't really need it.

**Review by cam-schultz — COMMENTED** [review]
Thanks for working through this and cleaning things up! This guide has definitely fallen through the cracks a bit as the contracts have evolved.

My only note is that the Solidity code blocks have odd breaks in them in a handful of places, which I've commented.

File: contracts/src/CrossChainApplications/GETTING_STARTED.md (hunk 2 above)
Comment (feuGeneA) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md:153]: the code blocks are "split" such that the bit that's to be copied for this particular instruction is in its own code block, whereas the surrounding "context" code, which is repeated from earlier instructions, is in is not included in the code block that needs to be copied to follow the instruction. this way the reader can easily hit the "copy" button on the code block needed to follow the instruction, without also copying the "context code" that they have already copied and pasted while following previous instructions.

i also explained that in the commit message [here](https://github.com/ava-labs/teleporter/pull/337/commits/fb8c1c045bcf61c3f136a5e26b8268d2605f6664). but i wrote the explanation for this comment fresh off the top of my head. i know the logic is a little complicated/convoluted, but hopefully _one_ of these explanations helps convey what i'm getting at. :)

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 8]
```diff
@@ -20,245 +20,227 @@ pragma solidity 0.8.18;
...
-    ReentrancyGuard,
-    ITeleporterReceiver
-{
     using SafeERC20 for IERC20;
-    ...
-}
 ```
 
-Then in `sendMessage` check whether `feeAmount` is greater than zero. If it is, transfer and approve the amount of IERC20 asset at `feeTokenAddress` to the Teleporter Messenger saved as a state variable.
+Then implement the `sendMessage` function with a check for whether `feeAmount` is greater than zero. If it is, transfer and approve the amount of IERC20 asset at `feeTokenAddress` to the Teleporter Messenger saved as a state variable.
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:148]:
```suggestion
Then add a check to the `sendMessage` function for whether `feeAmount` is greater than zero. If it is, transfer and approve the amount of IERC20 asset at `feeTokenAddress` to the Teleporter Messenger saved as a state variable.
```

The wording here suggests that the fee check fully implements the function, which it does not.

File: contracts/src/CrossChainApplications/GETTING_STARTED.md [hunk 9]
```diff
@@ -20,245 +20,227 @@ pragma solidity 0.8.18;
...
+                        amount: adjustedFeeAmount
+                    }),
+                    requiredGasLimit: requiredGasLimit,
+                    allowedRelayerAddresses: new address[](0),
+                    message: abi.encode(message)
+                })
+            );
 ```
 
 With the sending side complete, the next step is to implement `ITeleporterReceiver.receiveTeleporterMessage`. The receiver in this example will just receive the arbitrary string data, and check that the message is sent through Teleporter.
```
Comment (cam-schultz) [inline contracts/src/CrossChainApplications/GETTING_STARTED.md:197]:
```suggestion
With the sending side complete, the next step is to implement `ITeleporterReceiver.receiveTeleporterMessage`. The receiver in this example will just receive the arbitrary string data, and check that the message is sent through Teleporter. To  the `receiveTeleporterMessage` function, add:
```

File: contracts/src/CrossChainApplications/GETTING_STARTED.md (hunk 7 above)
Comment (cam-schultz) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md:294]: Do we want to remove this triple code block as well?

Comment (feuGeneA) [reply contracts/src/CrossChainApplications/GETTING_STARTED.md]: Ah yes indeed, thank you, done in https://github.com/ava-labs/teleporter/commit/3814571134a29106bf2d0ccb57fc149efd24a370

**Review by cam-schultz — APPROVED** [review]
LGTM!

**Pull request merged**